### PR TITLE
Ioctl template

### LINKED ISSUE #28: Create ioctl template file for driver
**Description**
This issue involves creating the template files necessary for defining the various ioctl codes

**Prerequisites**
#17 and #18 are completed and done

**Goal**
This ticket requires the following:
- [x] Create template file(s) for defining ioctl commands

**Definition of Done**
This ticket is completed when the device driver is set up such that ioctl codes and defining functions can be added in the future.

### PR DESCRIPTION
**Pull request by Rohanvamshi**
Added Ioctl function definition and implementation, added space for Ioctl command macro definitions in dev_gpio.h

### CODE REVIEW

File: Source_code/dev_gpio/include/dev_gpio.h [hunk 1]
```diff
@@ -0,0 +1,13 @@
+
+#ifndef dev_gpio_h
```
Comment (bama4) [inline Source_code/dev_gpio/include/dev_gpio.h]: Header guards for .h files are usually all caps for example:
`#ifndef DEV_GPIO_H`

File: Source_code/dev_gpio/include/dev_gpio.h [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+// define IOC commands 
+#define DEV_GPIO_IOC_MAGIC  'K'
+#define DEV_GPIO_IOC_RESET    _IO(GPIO_DEV_IOC_MAGIC, 0)
+
+
+
+
+
+#endif
```
Comment (bama4) [inline Source_code/dev_gpio/include/dev_gpio.h]: May want to make this 
`#endif /* DEV_GPIO_H */`

just to make it a bit clear what this statement is ending

**Comment by bama4** [comment]
@Rohanvamshi  Looks like the bulb, and bulb.o binary files were accidentally added.
Also, looks like some other commits got added accidentally.

The only commit that applies to this branch is 71429a3d9ece33ea8e2b67d4f5de9fdf492f45f1
To delete the other commits, you can do `git rebase -i master` to delete the extra commits that don't have the ioctl template changes.

**Comment by Rohanvamshi** [comment]
How did the older commits get there? That's weird They don't show up on my local git under the Ioctl branch. Ioctl branch has only the last 2 commits under it.

**Pull request merged**